extensions/amp-vk/0.1/amp-vk.js

```diff
@@ -0,0 +1,249 @@
+    this.iframeUrl_ = null;
+
+    /** @private {?Object} */
+    this.iframeParams_ = null;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+
+    /** @private {?number} */
+    this.widgetHeight_ = 0;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?string} */
+    this.embedType_ = null;
+
+    /** @private {?string} */
+    this.ownerId_ = null;
+
+    /** @private {?string} */
+    this.postId_ = null;
+
+    /** @private {?string} */
+    this.hash_ = null;
+
+    /** @private {?String} */
+    this.apiId_ = null;
+
+    /** @private {?String} */
+    this.pollId_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://vk.com', opt_onLayout);
+  }
+
+  /** @private */
+  getIFrameSrc_() {
+    const startWidth = this.element./*OK*/offsetWidth;
+    const pageUrl = this.element.ownerDocument
+        .location.href.replace(/#.*$/, '');
+    const pageReferrer = this.element.ownerDocument.referrer;
+    const createdTime = Number(new Date()).toString(16);
+
+    let q = '';
+    let src, queryParams;
+
+    if (this.embedType_ === embedTypes.POST) {
+      src = 'https://vk.com/widget_post.php';
+      queryParams = {
+        'app': 0,
+        'width': '100%',
+        '_ver': 1,
+        'owner_id': this.ownerId_,
+        'post_id': this.postId_,
+        'hash': this.hash_,
+        'amp': 1,
+        'startWidth': startWidth,
+        'url': pageUrl,
+        'referrer': pageReferrer,
+        'title': 'AMP Post',
+      };
+    } else if (this.embedType_ === embedTypes.POLL) {
+      src = 'https://vk.com/al_widget_poll.php';
+      queryParams = {
+        'app': this.apiId_,
+        'width': '100%',
+        '_ver': 1,
+        'poll_id': this.pollId_,
+        'amp': 1,
+        'url': pageUrl,
+        'title': 'AMP Poll',
+        'description': '',
+        'referrer': pageReferrer,
+      };
+    }
+
+    for (const param in queryParams) {
+      q += `&${param}=${encodeURIComponent(queryParams[param])}`;
+    }
+
+    src += `?${q.substr(1)}&${createdTime}`;
+
+    return src;
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(this.element.getAttribute('data-embedtype'),
+        'The data-embedtype attribute is required for <amp-vk> %s',
+        this.element);
+
+    this.embedType_ = this.element.getAttribute('data-embedtype');
+
+    if (this.embedType_ === embedTypes.POST) {
+      this.postBuildCallback_();
+    } else if (this.embedType_ === embedTypes.POLL) {
+      this.pollBuildCallback_();
+    }
+  }
+
+  /** @private */
+  postBuildCallback_() {
+    user().assert(this.element.getAttribute('data-hash'),
+        'The data-hash attribute is required for <amp-vk> Post %s',
+        this.element);
+
+    user().assert(this.element.getAttribute('data-owner_id'),
```

Comment:
could we rename all `data-x_id` to `data-xid`? I know it won't match your documentation but mixing `-` and `_` is strange.

Reply:
renamed `data-x_id` to `data-x-id` (i want to keep  '-' divider so that e.g. data-api-id looks readable)